service/history/shard/context_impl.go

```diff
@@ -1030,6 +1030,7 @@ func (s *ContextImpl) emitShardInfoMetricsLogsLocked() {
 	replicationLag := s.transferMaxReadLevel - s.shardInfo.ReplicationAckLevel
 	transferLag := s.transferMaxReadLevel - s.shardInfo.TransferAckLevel
 	timerLag := time.Since(timestamp.TimeValue(s.shardInfo.TimerAckLevelTime))
+	visibilityLag := s.transferMaxReadLevel - s.shardInfo.VisibilityAckLevel
```

Comment:
Just to clarify, the transferMaxReadLevel is the max taskID of all queues, and the lag here is just an approximate value, is that correct?

Reply:
yes, that's correct.